server/routerlicious/api-report/protocol-base.api.md

```diff
@@ -106,23 +130,42 @@ export function isSystemMessage(message: ISequencedDocumentMessage): boolean;
 export function mergeAppAndProtocolTree(appSummaryTree: ITree_2, protocolTree: ITree_2): ICreateTreeEntry[];
 
 // @public
-export class ProtocolOpHandler {
+export class ProtocolOpHandler implements IProtocolHandler {
```

Comment:
As a general rule, we should consider exporting only functor to create it, and not expose class itself. That's especially true with this work, though test code might want to look into guts.

Reply:
Would something like this be viable? https://github.com/microsoft/FluidFramework/pull/10412/commits/ca0194dcd9bbeca1c405848f7f8b4fd8b10d46b7#diff-e4758e313a7e2cbeafa89bd3c08d5e48ffc2b2e205183e913faddaa021b5cf55R58

but we'd need to export both of them (at least for now, maybe in `next` we can remove it) as removing this class from exports is breaking the contract.

Reply:
Sure, though I'd drop ProtocolHandlerBuilder (type) - the shape of that function can be different for different implementations (unless we have to do type erasure and pass functor to Container, not actual implementation of interface).

Reply:
Was thinking this to be part of the loader. This package should give us the interface, then the loader has its own builder signature which people can implement.